scipy/stats/_stats_py.py

```diff
@@ -84,7 +84,13 @@
            'brunnermunzel', 'alexandergovern']
 
 
+# This should probably be rewritten to avoid nested TypeErrors in favor of
+# branching based on dtype.
 def _contains_nan(a, nan_policy='propagate'):
+    # only inexact (floating/complexfloating) and object arrays support np.nan
+    if not (np.issubdtype(a.dtype, np.inexact)
+            or np.issubdtype(a.dtype, object)):
```

Comment:
@AtsushiSakai I agree that we don't need the `dtype` check. I didn't stop to think - `Series` and other array-like objects will have a `dtype` attribute, so we should be fine without the check.

Use of the built-in `sum` funciton instead of `.sum` method or `np.sum` is a different question, I think.  I'll address it in gh-16318, though.